### PR TITLE
Month sorting

### LINKED ISSUE #2: Top Month Always Empty (Archive Sidebar)
- When I click on the top-most month in the Archive Sidebar it says "No posts found...".
- When I create a post in the _current month_ (e.g. February) it displays a link in the sidebar for _next month_ (e.g. March), and says "No posts found..."

### LINKED ISSUE #3: Months sorted incorrectly (Archive Sidebar)
The sort order of months in the Archive Sidebar is off, with January 2015 appearing directly above December 2015:

archive sidebar sort order

### PR DESCRIPTION
**Pull request by jroers**
Fixes #3 
Fixes #2 

So these two are linked. By solving issue 3, it solves issue 2.

Since months are designated by their number (1-12), all the months except December would remain the same. Taking the months' values %12 and adding one caused the disorder. 1 becomes 2, 2 becomes 3... 12 would become 1.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -73,6 +73,7 @@ gem 'recaptcha', require: 'recaptcha/rails', branch: 'rails3'
 gem 'rubypants', '~> 0.2.0'
 gem 'twitter', '~> 5.14.0'
 gem 'uuidtools', '~> 2.1.1'
+gem 'psych', '~> 2.0', '>= 2.0.17'
```
Comment (tgaff) [inline Gemfile]: why do we need to add this to the gemfile?

File: lib/archives_sidebar/app/views/archives_sidebar/_content.html.erb [hunk 2]
```diff
@@ -1,11 +1,11 @@
 <% unless sidebar.archives.blank? %>
-  <h3 class="sidebar_title"><%= sidebar.title %></h3>
-  <div class="sidebar_body">
-    <ul id="archives">
```
Comment (tgaff) [inline lib/archives_sidebar/app/views/archives_sidebar/_content.html.erb:4]: are you sure this ID isn't used by some other code ?  (JS, RSpec, etc)

File: lib/archives_sidebar/app/views/archives_sidebar/_content.html.erb [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+  <h3 class="sidebar-title"><%= sidebar.title %></h3>
```
Comment (nathanallen) [inline lib/archives_sidebar/app/views/archives_sidebar/_content.html.erb:2]: How is this html change relevant to issue #2?

**Comment by nathanallen** [comment]
Good explanation!
